static/js/src/modules/search-panel.js

```diff
@@ -0,0 +1,45 @@
+const searchPanel = () => {
+  /**
+    Sets up the search handler.
+  */
+  const form = document.querySelector('[data-js="search-form"]'),
+    input = document.querySelector('[data-js="form-text"]'),
+    header = document.querySelector('[data-js="navigation"]'),
+    searchClose = document.querySelector('[data-js="search-close"]');
+
+  input.addEventListener('focus', e => {
+    header.classList.add('search-active');
+  });
+
+  if (searchClose !== null) {
+    searchClose.addEventListener('click', e => {
+      e.preventDefault();
+      header.classList.remove('search-active');
+      input.value = '';
+    });
+  }
+
+  window.addEventListener('keydown', e => {
+    if (e.keyCode === 27) {
+      input.blur();
+    }
+  });
+
+  form.addEventListener('submit', e => {
+    e.preventDefault();
+    const text = document.querySelector('input[name="text"]').value;
+    const re = new RegExp(' ', 'g');
+    const url = '/q/' + text.replace(re, '/');
+    window.location.pathname = url;
```

Comment:
Why do we do this instead of just have the form submit naturally?

Reply:
Oops, we don't need this, my mistake when copying this over from blues_browser.